src/@next/TextArea/TextArea.tsx

```diff
@@ -20,82 +19,93 @@ export type TextAreaProps = Omit<
   onChange: (value: string) => void;
   error?: boolean;
   width?: string;
+  forwardedRef?: RefObject<HTMLTextAreaElement>;
+  /**
+   * if true, allows the user to type more than the maxLength.
+   * if false, the user will not be able to type more than the maxLength,
+   * all the characters typed after the maxLength will be ignored.
+   *
+   * **defaults to** `true`
+   */
+  canExceedMaxLength: boolean;
 };
 
-export const TextArea = React.forwardRef<HTMLTextAreaElement, TextAreaProps>(
-  function TextArea(
-    {
-      value,
-      rows = 3,
-      maxLength,
-      error = false,
-      disabled = false,
-      width = '520px',
-      onChange,
-      ...props
-    }: TextAreaProps,
-    ref
-  ) {
-    const [charCount, setCharCount] = useState<number>(0);
-    const [isFocused, setIsFocused] = useState<boolean>(false);
-    const isError: boolean = error || (!!maxLength && charCount > maxLength);
+const _TextArea = ({
+  value,
+  rows = 3,
+  maxLength,
+  error = false,
+  disabled = false,
+  width = '520px',
+  onChange,
+  forwardedRef,
+  canExceedMaxLength = true,
+  ...props
+}: TextAreaProps) => {
+  const [isFocused, setIsFocused] = useState<boolean>(false);
+  const charCount = value.length;
 
-    const localRef = useRef<HTMLTextAreaElement>(null);
-    const setRefs = (node: HTMLTextAreaElement) => {
-      localRef.current = node;
-      if (typeof ref === 'function') {
-        (ref as (instance: HTMLTextAreaElement | null) => void)(node);
-      } else if (ref) {
-        (ref as MutableRefObject<HTMLTextAreaElement | null>).current = node;
-      }
-    };
+  const transformedMaxLength: number = maxLength || undefined;
```

Comment:
do you still need this? if you don't pass maxLength, won't the value already be undefined?

Reply:
I think he needs that when maxLength is 0

Reply:
@ninariccimarie reason for adding this was to avoid confusion, for some reason when i hover on the maxLength it always shows as `number` which leads to confusion as to why its always defined, let me know if we show remove this

Reply:
@Afsal102 you can remove this. I verified that when you start typing to pass maxLength in TextArea component, it will show you the correct type.
<img width="528" alt="Screenshot 2023-07-05 at 11 49 59 AM" src="https://github.com/glints-dev/glints-aries/assets/7514754/0194e84c-de2b-41cc-a47f-0cdd3c6fe2f4">

engineers can verify through the TextAreaProps as well from the TextareaHTMLAttributes

Reply:
@ninariccimarie but when u are working inside the TextArea implementation, this doesnt show it like this! nevertheless  i agree with the second point

Reply:
![Uploading Screenshot 2023-07-05 at 09.32.26.png…]()